websocket/server.go

```diff
@@ -333,14 +336,18 @@ func (s *server) broadcastMessageToTopic(ctx context.Context, m MessageForBroadc
 			srv.Close()
 			return
 
-		// [CASE] Send the message!
-		case client.send <- broadcastMessage.Message:
-			logrus.Infof("broadcastMessageToTopic: send")
-
-		// [CASE] Unresponsive client.
+			// [CASE] Send message!
 		default:
-			logrus.Infof("broadcastMessageToTopic: unregister due to unresponsiveness")
-			s.actuateUnregisterClient(client, errUndeliverableClient)
+			ok := client.SendMessage(broadcastMessage.Message)
```

Comment:
Because we no longer have access to client internal variables, then we use this new method that returns a boolean that indicates whether sending the message was successful. If it's not successful, we call `actuateUnregisterClient`.